boto3/resources/factory.py

```diff
@@ -352,34 +335,26 @@ def _create_class_partial(factory_self, resource_cls, identifiers=None):
         def create_resource(self, *args, **kwargs):
             pargs = []
 
+            # We lazy-load the class to handle circular references.
+            resource_cls = factory_self.load_from_definition(
+                service_name, name, resource_defs.get(name, {}),
+                resource_defs, service_model)
+
             # Assumes that identifiers are in order, which lets you do
             # e.g. ``sqs.Queue('foo').Message('bar')`` to create a new message
             # linked with the ``foo`` queue and which has a ``bar`` receipt
             # handle. If we did kwargs here then future positional arguments
             # would lead to failure.
+            identifiers = subresource.resource.identifiers
             if identifiers is not None:
-                for key, value in identifiers.items():
-                    pargs.append(getattr(self, xform_name(key)))
+                for identifier, value in build_identifiers(identifiers, self):
+                    pargs.append(value)
```

Comment:
Not a big deal, but I prefer the vars spelled out to avoid confusion, `positional_args` would work for me.